components/auth/src/main/java/org/cloudfoundry/credhub/config/auth/OAuth2AuthenticationExceptionHandler.java

```diff
@@ -0,0 +1,46 @@
+package org.cloudfoundry.credhub.config.auth;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.oauth2.core.OAuth2AuthenticationException;
+import org.springframework.security.oauth2.core.OAuth2Error;
+import org.springframework.security.oauth2.server.resource.web.BearerTokenAuthenticationEntryPoint;
+import org.springframework.security.web.AuthenticationEntryPoint;
+
+import com.nimbusds.jose.shaded.gson.JsonObject;
+
+/**
+ * Customized BearerTokenAuthenticationEntryPoint to fill the response body
+ * with json error data when an OAuth2AuthenticationException occurs.
+ */
+public class OAuth2AuthenticationExceptionHandler
+        implements AuthenticationEntryPoint {
+
+    @Override
+    public void commence(HttpServletRequest request,
+                         HttpServletResponse response,
+                         AuthenticationException authException)
+            throws IOException {
+        BearerTokenAuthenticationEntryPoint btaep =
+                new BearerTokenAuthenticationEntryPoint();
```

Comment:
Would it work to instantiate once as a bean and re-use rather than creating a new instance each commence()?

Reply:
Oops. That was oversight. Thank you for catching that.

Reply:
Fixed it with the new commit. Also marking this PR as ready to merge as you suggested in Gspace. Will do the additional tasks mentioned above as part of the associated story.

Reply:
lgtm